File: NumbersAppearOnce/NumbersAppearOnce.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ans = 0;
        for (int i = 0; i < 32; i++)
        {
            int count = 0;
            for (auto n : nums)
            {
                if ((1 << i ) & n) 
                    count++;
            }
            if (count % 3) 
                ans += (1 << i);
        }
        return ans;
    }
};

File: NumbersAppearOnce/NumbersAppearOnce.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        res = 0
        for i in range(0,32):
            count = 0
            for num in nums:
                if (num&(1<<i)):
                    count+=1
            if (count%3):
                res += 1<<i
        return res
